nuxeo-compound-documents/src/main/java/org/nuxeo/compound/documents/CompoundDocumentServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+                session.createDocumentModel(parent, compoundDocName, COMPOUND_DOCTYPE));
+        createCompoundDocument(compoundDoc, archive);
+        return compoundDoc;
+    }
+
+    protected boolean isAllowedEntry(ZipEntry entry) {
+        CompoundDocumentManager manager = Framework.getService(CompoundDocumentManager.class);
+        Path path = new Path(entry.getName());
+        // TODO Dynamically getIgnoredFiles depending on the compound type.
+        IgnoredFilesDescriptor ignoredFiles = manager.getIgnoredFiles(DEFAULT_IGNORED_FILES);
+        for (String segment : path.segments()) {
+            if (ignoredFiles.getIgnoredFileDescriptors().stream().anyMatch(i -> i.ignore(segment))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    protected void createCompoundDocument(DocumentModel compoundDoc, Blob archive) {
+        try (ZipFile zip = new ZipFile(archive.getFile())) {
+            FileManagerService fileManager = (FileManagerService) Framework.getService(FileManager.class);
+            zip.stream().filter(this::isAllowedEntry).sorted(Comparator.comparing(ZipEntry::getName)).forEach(entry -> {
+                Path entryPath = new Path(entry.getName());
+                String parentPath = compoundDoc.getPathAsString() + separator
+                        + entryPath.removeLastSegments(1).toString();
+                if (entry.isDirectory()) {
+                    CoreSession session = compoundDoc.getCoreSession();
+                    session.createDocument(
+                            session.createDocumentModel(parentPath, entryPath.lastSegment(), COMPOUND_FOLDER_DOCTYPE))
+                           .getPathAsString();
+                } else {
+                    ZipEntryBlob blob = new ZipEntryBlob(zip, entry);
+                    blob.setFilename(new Path(entry.getName()).lastSegment());
+                    FileImporterContext ctx = FileImporterContext.builder(compoundDoc.getCoreSession(), blob,
+                            parentPath).build();
+                    try {
+                        fileManager.createOrUpdateDocument(ctx);
+                    } catch (IOException e) {
+                        String message = String.format(
+                                "Failed to create CompoundDocument: %s. something went wrong with archive: %s at entry: %s.",
+                                compoundDoc.getName(), archive.getFilename(), entry.getName());
+                        log.warn(message, e);
+                        throw new NuxeoException(message, e);
+                    }
+                }
+            });
+        } catch (IOException e) {
+            String message = String.format(
+                    "Failed to create CompoundDocument: %s. something went wrong with archive: %s.",
+                    compoundDoc.getName(), archive.getFilename());
+            log.warn(message, e);
+            throw new NuxeoException(message, e);
```

Comment:
Either log or throw, do not do both.

Reply:
remove the message variable and format on the fly as I do